Assets/Mirror/Editor/Weaver/CompilationFinishedHook.cs

```diff
@@ -49,32 +49,29 @@ static void HandleError(string msg)
         [InitializeOnLoadMethod]
         static void OnInitializeOnLoad()
         {
-            CompilationPipeline.assemblyCompilationFinished += OnCompilationFinished;
-
             // We only need to run this once per session
             // after that, all assemblies will be weaved by the event
             if (!SessionState.GetBool("MIRROR_WEAVED", false))
             {
+                // reset session flag
                 SessionState.SetBool("MIRROR_WEAVED", true);
+
                 WeaveExistingAssemblies();
             }
+
+            // hook the event after the first run of WeaveExistingAssemblies to avoid a double firing
+            CompilationPipeline.assemblyCompilationFinished += OnCompilationFinished;
```

Comment:
Why did this get moved here?   are you sure it is safe to move it below WeaveExistingAssemblies?

Reply:
I didn't want the event to be somehow fired by the first pass, causing a 2nd pass.